src/primitive-field/primitive-field.component.ts

```diff
@@ -71,12 +72,10 @@ export class PrimitiveFieldComponent extends AbstractFieldComponent implements O
   }
 
   commitValueChange() {
-    // Validation
     let errors = this.schemaValidationService.validateValue(this.value, this.schema);
-    if (!errors.length) {
-      this.jsonStoreService.setIn(this.path, this.value);
-    }
-    this.errors = errors;
+    this.jsonStoreService.setIn(this.path, this.value);
+    this.internalErrors = errors;
+    this.appGlobalsService.internalErrors = {[this.pathString]: errors};
```

Comment:
A better naming can be `this.appGlobalsService.internalErrorMap` and using a setter makes it hard to read, since you cannot realise that some extra steps (such as merging the map) are happening.